### PR TITLE
feat(devcontainer): vscode devcontainer configuration

### PR DESCRIPTION
**Pull request by valter-silva-au**
Add VSCode DevContainer configuration to the project. The goal is to ease development by including a systematic way of consistently having all the dependencies installed and ready. This pull-request achieves a similar goal as described in https://github.com/aws/aws-pdk/issues/603.

Artifacts:
```bash
node ➜ /workspaces/aws-pdk (feat/devcontainer) $ devcontainer-info 

Development container image information

- Image version: 1.0.8
- Definition ID: typescript-node
- Variant: 18-bullseye
- Source code repository: https://github.com/devcontainers/images
- Source code release/branch: v0.3.27
- Timestamp: Wed, 10 Jan 2024 19:54:32 GMT

More info: https://github.com/devcontainers/images/tree/main/src/typescript-node/history/1.0.8.md

node ➜ /workspaces/aws-pdk (feat/devcontainer) $ pnpm --version
8.14.0
node ➜ /workspaces/aws-pdk (feat/devcontainer) $ poetry --version
Poetry (version 1.7.1)
node ➜ /workspaces/aws-pdk (feat/devcontainer) $ projen --version
0.79.1
node ➜ /workspaces/aws-pdk (feat/devcontainer) $ java --version
openjdk 11.0.21 2023-10-17 LTS
OpenJDK Runtime Environment Microsoft-8519785 (build 11.0.21+9-LTS)
OpenJDK 64-Bit Server VM Microsoft-8519785 (build 11.0.21+9-LTS, mixed mode)
```

Reference:
- https://code.visualstudio.com/docs/devcontainers/containers
- Base Image: https://github.com/devcontainers/images/tree/main/src/typescript-node
- Available features: https://containers.dev/features

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "Node.js 18 & TypeScript",
+  "image": "mcr.microsoft.com/devcontainers/typescript-node:1-18-bullseye",
+  "features": {
+    "ghcr.io/devcontainers-contrib/features/projen:1": {}
```
Comment (agdimech) [inline .devcontainer/devcontainer.json:16]: Don't we need in include pnpm here?

Additionally, do we need to install mvn, jdk, python and poetry also?

**Review by agdimech — COMMENTED** [review]
Great contribution - added some comments.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (valter-silva-au) [reply .devcontainer/devcontainer.json:16]: `pnpm 8.14.0` and `python3.9.2` is already installed with the base image. I can include JDK, and Poetry if you like @agdimech. Would look like this:

```json

  "ghcr.io/devcontainers/features/java": {
      "installGradle": true,
      "installMaven": true,
      "version": "11"
  },
  "ghcr.io/devcontainers-contrib/features/pnpm": {
      "version": "8.11.0"
  },
  "ghcr.io/devcontainers-contrib/features/poetry":{
      "version": "latest"
  },
  ```

Comment (agdimech) [reply .devcontainer/devcontainer.json:16]: Yes that would be amazing! Thank you :)

We could potentially make this a feature of the monorepo itself also so consumers of the PDK monorepo can have this generated and managed. Something like:

`
const monorepo = new MonorepoTsProject({
    name: "my-project",
    packageManager: javascript.NodePackageManager.PNPM,
    vsCode: {
       devContainer: true,
    }
});
`

Then the generated `devcontainer.json` would be managed by projen and if the update these deps in future, consumers can just re-synth :) What do you think?

Comment (valter-silva-au) [reply .devcontainer/devcontainer.json:16]: That's a great idea @agdimech!

Comment (valter-silva-au) [reply .devcontainer/devcontainer.json:16]: @agdimech requirements included.

**Review by agdimech — APPROVED** [review]
Great work getting this added, i'm excited to have a play around with it!

In terms of adding this to the Monorepo construct, we can track that in: https://github.com/aws/aws-pdk/issues/603.

**Pull request merged**